### PR TITLE
Blacklist

### PR DESCRIPTION
**Pull request by kimlai**
A user can blacklist a track, which means that it'll never re-appear in
the user's feed.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -76,4 +71,15 @@ function *callback() {
     this.redirect('/');
 }
 
+function *blacklist() {
+    var token = this.state.token;
+    var soundcloudUserId = this.state.user.id;
+    yield knex .insert({
```
Comment (Sinewyk) [inline server.js]: dafuq whitespace between `knex` and `.insert` ?

File: server/feed.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+var knexfile = require('../knexfile');
+var knex = require('knex')(knexfile);
+var _ = require('lodash');
+var Promise = require('bluebird');
+
+module.exports = function fetchFeedApi(soundcloudUserId, token, nextLink) {
+    return Promise.join(
+        fetchSoundcloudFeed(token, nextLink),
+        fetchBlacklist(soundcloudUserId),
+        function (feed, blacklist) {
```
Comment (Sinewyk) [inline server/feed.js]: Hu ... Promise.join doesn't work like that afaik. You are using it like you're in nuclear-js ...

File: server/feed.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            feed.tracks = _.filter(feed.tracks, function (track) {
+                return typeof track !== 'undefined' && !_.includes(blacklist, track.id);
```
Comment (Sinewyk) [inline server/feed.js]: is the typeof really necessary ?

Comment (kimlai) [inline server/feed.js]: I hope it's not, couldn't figure out why I get some undefined value here. Will investigate.

File: server/feed.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+function fetchBlacklist(soundcloudUserId) {
+    return knex
+        .select('soundcloudTrackId')
+        .from('blacklist')
+        .where({soundcloudUserId: soundcloudUserId})
+        .then(function (rows) {
+            return _.map(rows, function (row) {
+                return parseInt(row.soundcloudTrackId);
```
Comment (Sinewyk) [inline server/feed.js]: `return parseInt(row.soundcloudTrackId, 10);`

File: server/feed.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+var knexfile = require('../knexfile');
+var knex = require('knex')(knexfile);
+var _ = require('lodash');
+
+module.exports = function fetchFeedApi(soundcloudUserId, token, nextLink) {
+    return Promise.all([
+            fetchSoundcloudFeed(token, nextLink),
+            fetchBlacklist(soundcloudUserId),
+        ])
+        .then(function (results) {
```
Comment (Sinewyk) [inline server/feed.js]: spread operator

**Pull request merged**